### PR TITLE
Token allowance unit tests

### LINKED ISSUE #16910: Add unit tests for token allowance
It seems that currently token-allowance has 0% test coverage, we should create and add unit tests for it.

### PR DESCRIPTION
**Pull request by amerkadicE**
## Explanation

This PR adds unit tests for the token allowance screen.

* Fixes #16910 

## Pre-merge author checklist

- [x] I've clearly explained:
  - [x] What problem this PR is solving
  - [x] How this problem was solved
  - [x] How reviewers can test my changes
- [ ] Sufficient automated test coverage has been added

## Pre-merge reviewer checklist

- [ ] Manual testing (e.g. pull and build branch, run in browser, test code being changed)
- [ ] PR is linked to the appropriate GitHub issue
- [ ] **IF** this PR fixes a bug in the release milestone, add this PR to the release milestone

If further QA is required (e.g. new feature, complex testing steps, large refactor), add the `Extension QA Board` label.

In this case, a QA Engineer approval will be be required.

### CODE REVIEW

File: ui/pages/token-allowance/token-allowance.test.js [hunk 1]
```diff
@@ -0,0 +1,332 @@
...
+      userFeeLevel: 'custom',
+      defaultGasEstimates: {
+        estimateType: 'custom',
+        gas: '0xea60',
+        maxFeePerGas: '0x4a817c800',
+        maxPriorityFeePerGas: '0x4a817c800',
+        gasPrice: '0x4a817c800',
+      },
+    },
+  };
```
Comment (jpuri) [inline ui/pages/token-allowance/token-allowance.test.js:250]: Nice work on creating configuration props for the component 👍

File: ui/pages/token-allowance/token-allowance.test.js [hunk 2]
```diff
@@ -0,0 +1,332 @@
...
+    fireEvent.click(nextButton);
+
+    const backButton = getByText('< Back');
+    fireEvent.click(backButton);
+
+    expect(getByText('Set a spending cap for your')).toBeInTheDocument();
+  });
+
+  it('should click Verify contract details and show popup Contract details', () => {
+    const store = configureMockStore()(state);
```
Comment (jpuri) [inline ui/pages/token-allowance/token-allowance.test.js:317]: May be can be moved to `beforeEach`

```
const store = configureMockStore()(state);
```

**Review by jpuri — DISMISSED** [review]

File: ui/pages/token-allowance/token-allowance.test.js [hunk 3]
```diff
@@ -0,0 +1,332 @@
+import React from 'react';
+import configureMockStore from 'redux-mock-store';
+import { fireEvent } from '@testing-library/react';
+import { renderWithProvider } from '../../../test/lib/render-helpers';
+import { CHAIN_IDS } from '../../../shared/constants/network';
+import TokenAllowance from './token-allowance';
+
+const testTokenAddress = '0xC011a73ee8576Fb46F5E1c5751cA3B9Fe0af2a6F';
+const state = {
```
Comment (digiwand) [inline ui/pages/token-allowance/token-allowance.test.js:9]: I wonder if mockState (`import mockState from '../../../../test/data/mock-state.json';`) would have been sufficient

Comment (digiwand) [inline ui/pages/token-allowance/token-allowance.test.js:9]: I wonder if `mockState` from `../../../../test/data/mock-state.json` would have been sufficient here

**Review by digiwand — DISMISSED** [review]
Nice!

**Comment by jpuri** [comment]
@amerkadicE : can you plz check build.

**Review by digiwand — DISMISSED** [review]

File: ui/pages/token-allowance/token-allowance.test.js (hunk 3 above)
Comment (jpuri) [reply ui/pages/token-allowance/token-allowance.test.js:9]: That will be useful to remove redundancy.

File: ui/pages/token-allowance/token-allowance.test.js [hunk 4]
```diff
@@ -0,0 +1,334 @@
...
+        value: '0x0',
+        data: '0x095ea7b30000000000000000000000009bc5baf874d2da8d216ae9f137804184ee5afef40000000000000000000000000000000000000000000000000000000000011170',
+        gas: '0xea60',
+        gasPrice: '0x4a817c800',
+        maxFeePerGas: '0x4a817c800',
+      },
+      origin: 'https://metamask.github.io',
+      type: 'approve',
+      history: [
+        {
```
Comment (jpuri) [inline ui/pages/token-allowance/token-allowance.test.js:179]: `history` here can be removed.

File: ui/pages/token-allowance/token-allowance.test.js [hunk 5]
```diff
@@ -0,0 +1,334 @@
...
+    );
+
+    const verifyContractDetails = getByText('Verify contract details');
+    fireEvent.click(verifyContractDetails);
+
+    expect(getByText('Contract details')).toBeInTheDocument();
+
+    const gotIt = getByText('Got it');
+    fireEvent.click(gotIt);
+    expect(verifyContractDetails).toBeInTheDocument();
```
Comment (jpuri) [inline ui/pages/token-allowance/token-allowance.test.js:332]: Above test is not checking for popup. `expect(verifyContractDetails).toBeInTheDocument();` does not looks useful.

File: ui/pages/token-allowance/token-allowance.test.js (hunk 4 above)
Comment (amerkadicE) [reply ui/pages/token-allowance/token-allowance.test.js]: Done

File: ui/pages/token-allowance/token-allowance.test.js (hunk 5 above)
Comment (amerkadicE) [reply ui/pages/token-allowance/token-allowance.test.js]: I updated the test to check if pop-up is closed when `Got it` button is clicked.

File: ui/pages/token-allowance/token-allowance.test.js (hunk 3 above)
Comment (amerkadicE) [reply ui/pages/token-allowance/token-allowance.test.js:8]: I tried using mockState, but some properties are missing and the `gasEstimateType` property needs a different value. 
That is why I created a custom state in the test file.

**Review by darkwing — APPROVED** [review]

File: coverage-targets.js [hunk 6]
```diff
@@ -6,10 +6,10 @@
...
   global: {
-    lines: 63,
-    branches: 51,
-    statements: 62.45,
-    functions: 55.5,
+    lines: 63.5,
+    branches: 52,
+    statements: 62.75,
+    functions: 56,
   },
```
Comment (jpuri) [inline coverage-targets.js:13]: 👍

**Review by jpuri — APPROVED** [review]

**Pull request merged**